kqueen/engines/openstack_kubespray.py

```diff
@@ -569,19 +568,18 @@ def _run_ansible(self, inventory="hosts.json", playbook="cluster.yml"):
             "--extra-vars", "docker_dns_servers_strict=no",
         ]
         env = self._construct_env()
-        # TODO(sskripnick) Maybe collect out/err from pipe and log them
-        # separately.
-        pipe = subprocess.Popen(
-            args,
-            stdin=subprocess.DEVNULL,
-            stdout=sys.stdout,
-            stderr=sys.stderr,
-            cwd=self.kubespray_path,
-            env=env,
-        )
-        pipe.wait()
-        if pipe.returncode:
-            logger.warning("Non zero exit status from ansible (%s)" % pipe.returncode)
+        with open(os.path.join(self._get_cluster_path(), "ansible_log.txt"), "a+") as ansible_log:
```

Comment:
why we can't log it into common log?

Reply:
it is ansible entity, we are logging only if something went wrong